Assets/UXF/Scripts/Trackers/ScreenRayTracker.cs

```diff
@@ -0,0 +1,150 @@
+using UnityEngine;
+using System.Collections;
+using System.Collections.Generic;
+using UXF;
+using System.Linq;
+
+/// <summary>
+/// Attach this component to any gameobject (e.g. an empty one) and assign it in the trackedObjects field in an ExperimentSession to record
+/// if ray casted from camera is hitting anything. NOTE: Update Type must be set to MANUAL. 
+/// </summary>
+public class ScreenRayTracker : Tracker {
+	// Public vars
+    [Header("Necessary Input")]
+    public Camera cam;
+    public Session session;
+
+    [Header("Ray coordinates")]
+    [TextArea(10, 10)]
+    public string HowToUse = "Please provide coordinates in form of lists named ray_x & ray_y in your .json file. For example \n\"ray_x\": [0.5],\n\"ray_y\": [0.5]\nif you want to have one ray in the middle of the screen. Multiple rays can be provide with this method.";
+
+    [Header("Optional Input")]
+    [Tooltip("Enable if you want to visualise the rays in the scene view and the console output.")]
+    public bool debugMode = true; 
+    [Tooltip("The max distance the ray should check for collisions. For further information see manual of Physics.Raycast.")]
+    public float distance = Mathf.Infinity;
+
+    [Tooltip("Set to true if you want to use a LayerMask for the rays (see maunual of LayerMask.GetMask). Note you also need to set Layer Mask Names in this case with one or more layers that you want to use.")]
+    public bool useLayerMask = false;
+    [Tooltip("Provide the names of the layers for the mask.")]
+    public string[] layerMaskNames;
+
+    // Private vars
+    private List<string> objectDetected = new List<string>();
+    private UXFDataRow currentRow;
+    private bool recording = false;
+    private int layerMask;
+    private string noObjectString = "NA";
+    private int numRays;
+    private List<float> x  = new List<float>();
+    private List<float> y  = new List<float>();
+
+    // Start calc
+    void Start(){
+        // Create layer mask 
+        if(useLayerMask){
+            layerMask = LayerMask.GetMask(layerMaskNames);
+        } else {
+            layerMask = ~0; // Set to everything as no mask is wanted. 
+        }
+
+    	// Start the recoding
+        StartCoroutine(RecordRoutine());
+    }
+
+    /// <summary>
+    /// Gets coordinates for the rays in screen space from .json file and prints screen resolution
+    /// </summary>
+    public void GetRayCoordinates(){
+        // Coordinates
+        x = session.settings.GetFloatList("ray_x");
+        y = session.settings.GetFloatList("ray_y");
+
+        // Screen resolution
+        Debug.Log(Screen.currentResolution);
+    }
+
+    /// <summary>
+    /// Starts the recording. This method needs to be added to [UXF_Rig] events called On Trial Begin
+    /// </summary>
+    public void StartRecording(){
+    	recording = true;
+    }
+
+    /// <summary>
+    /// Stops the recording. This method needs to be added to [UXF_Rig] events called On Trial End
```

Comment:
why does it need to be added to `On Trial End`? trackers have start and stop recording methods